### PR TITLE
Add queries for SI metrics

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Related-To: https://github.com/thoth-station/metrics-exporter/issues/474

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

Add queries for SI metrics that will be used by metrics-exporter and for SLO-reporter

### CODE REVIEW

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -1592,6 +1592,53 @@ def get_unsolved_python_package_versions_count_all(
...
+
+        query = query.filter(exists().where(and_(*conditions)))
+
+        return query
+
+    def get_si_analyzed_python_package_versions_count_all(
+        self,
+        index_url: Optional[str] = None,
+        *,
+        distinct: bool = False,
```
Comment (saisankargochhayat) [inline thoth/storages/graph/postgres.py:1622]: Should distinct be true by default?

Comment (pacospace) [reply thoth/storages/graph/postgres.py:1622]: we have False everywhere by default, as you want to count records in the database first, then if you want you can filter out :)

**Review by saisankargochhayat — APPROVED** [review]
LGTM :tada:

**Comment by pacospace** [comment]
@harshad16

**Review by fridex — APPROVED** [review]
lgtm :+1:

**Pull request merged**